File: Algorithms/763. Partition Labels/golang/solution.go
func partitionLabels(s string) []int {
	last := make(map[rune]int)
	for i, c := range s {
		last[c] = i
	}

	var anchor int = -1
	var j int
	var ans []int

	for i, c := range s {
		if last[c] > j {
			j = last[c]
		}

		if i == j {
			ans = append(ans, i-anchor)
			anchor = i
		}
	}
	return ans
}

File: Algorithms/763. Partition Labels/javascript/solution.js
/**
 * @param {string} s
 * @return {number[]}
 */
var partitionLabels = function (s) {
    const last = {};
    for (let i = 0; i < s.length; i++) {
        last[s[i]] = i;
    }
    let j = 0,
        anchor = -1;
    const ans = [];
    for (let i = 0; i < s.length; i++) {
        j = Math.max(j, last[s[i]]);
        if (i === j) {
            ans.push(i - anchor);
            anchor = i;
        }
    }
    return ans;
};
